src/lib/addons/slack.ts

```diff
@@ -73,8 +73,9 @@ export default class SlackAddon extends Addon {
             }
         }
 
-        const { text, url: featureLink } = this.msgFormatter.format(event);
-
+        const { text: formattedMessage, url: featureLink } =
+            this.msgFormatter.format(event);
+        const text = formattedMessage.substring(0, 3000);
```

Comment:
Just curious: Is this implementation simpler because this integration is deprecated?

Reply:
Simpler than what other implementation? I did it here because the "msgFormatter" is kind of generic, it doesn't say anything about Slack, so just for precaution I decided not to modify it. I was thinking "If this is a message formatter inside a Slack folder package by feature, then this might be only for Slack", but because it was not I just err on the safe side.

Reply:
Simpler than the one in Slack App. In my head it should be pretty much the same thing, so I was just curious why it wasn't.

Reply:
Oh, just because the slack-app is using the formatted text also as output for the event, this one is only using it for the push

Reply:
There's still a:

```ts
this.registerEvent({
    integrationId,
    state,
    stateDetails: stateDetails.join('\n'),
    event: serializeDates(event),
    details: {
        url,
        channels: slackChannels,
        username,
        message: text,
    },
});
```

here, which means that there's a small inconsistency between Slack and Slack App in the way they register integration events: Slack App will tell us the text was trimmed to 3000 characters, but Slack will just trim it.

It's a small detail, so I think it's fine, but again, was just curious 😄

Reply:
Oh! Thanks! I didn't understand. This is the PR for consistency:  https://github.com/Unleash/unleash/pull/8230

Reply:
I just didn't see the message at the bottom